src/classes/RD2_EnablementDelegate_CTRL.cls

```diff
@@ -128,15 +157,48 @@ public with sharing class RD2_EnablementDelegate_CTRL extends STG_Panel {
         return state;
     }
 
+    /***
+    * @description Determines if the dry run is outdated
+    * @param state Enablement state
+    * @return Boolean
+    */
+    private static Boolean isDryRunOutdated(EnablementState state) {
+        return batchJobService.getCompletedDaysBetween('RD2_DataMigrationDryRun_BATCH') > state.dryRunLimit;
+    }
+
+    /***
+    * @description Retrieves number of days that dry run is valid after completion
+    * @param state Enablement state
+    * @return Integer
+    */
+    private static Integer getDryRunLimit(EnablementState state) {
+        Integer dryRunLimit = DRY_RUN_VALID_DAYS;//default
+
+        //if this is a scratch org, enable override for testing purposes
+        if (UTIL_Namespace.isScratchOrg()) {
```

Comment:
@force2b , the dry run days valid value is not configurable, but to allow testing without waiting 8 days, the 'dryRunLimit` can be added/overwritten in the RD Settings "Recurring Donations Enablement State" and that will be taken into consideration when RD2 enablement page is loaded on a scratch org. The number needs to be > 0 in order to be accepted. Overwriting is not applicable on an org that is not a scratch org. 

Days between is number of the days between validation batch CompletedDate and today. The validation is outdated when days between is > 7 (or any overwritten value specified in the RD settings enablement state).

@ravensisco , @heidmary  ^

Reply:
Trying to test this out and I found a way to test without waiting a full day. If you remove the `&& state.dryRunLimit > 0` condition in the method below then it's possible to set the dryRunLimit to -1 in Custom Settings. That forces a run of today to be out of date. I don't believe this is an issue to change in the production code as it's a value that even if a customer were to set would still force them to re-run the validation. I think this would greatly help in testing/qa.

`{"isReady":true,"isMigrationEnabled":false,"isMetaLaunched":false,"isMetaConfirmed":false,"isEnabled":false,"isDryRun2":false,"isConfirmed":false,"dryRunLimit":-1}`

Reply:
Ah. That downside of the above is that the dryRunLimit is never reset back to 7. It stays as -1 so even after re-running the validation it still shows as out of date. For my testing, I just edited the custom setting to change -1 to 1.

Reply:
That is correct, settings it to -1 will outdate every dry run.